backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -242,24 +253,42 @@ public Person updateMe(
         gender,
         residentCongregateSetting,
         employedInHealthcare);
+    upsertPreferredLanguage(toUpdate, preferredLanguage);
     return _repo.save(toUpdate);
   }
 
+  public PatientPreferences getPatientPreferences(Person person) {
+    return _prefRepo.findByPerson(person).orElseGet(() -> new PatientPreferences(person));
+  }
+
   @AuthorizationConfiguration.RequirePermissionEditPatientAtFacility
-  public Person updateTestResultDeliveryPreference(
-      UUID patientId, TestResultDeliveryPreference testResultDelivery) {
-    Person toUpdate = _repo.findById(patientId).orElseThrow();
-    toUpdate.setTestResultDelivery(testResultDelivery);
-    return _repo.save(toUpdate);
+  public PatientPreferences updateTestResultDeliveryPreference(
+      UUID personId, TestResultDeliveryPreference testResultDelivery) {
+    Person person = _repo.findById(personId).orElseThrow();
+    return upsertTestResultDeliveryPreference(person, testResultDelivery);
   }
 
   // IMPLICIT AUTHORIZATION: this fetches the current patient after a patient link
   // is verified, so there is no authorization check
-  public Person updateMyTestResultDeliveryPreference(
+  public PatientPreferences updateMyTestResultDeliveryPreference(
```

Comment:
No updateMyPreferredLanguage?

Reply:
Correct – preferred language updates happen via `PersonService.updateMe`, whereas test result delivery preferences are set in AoE questions. I'm going to add a tech debt ticket to clear this up, because (1) it's obfuscated and confusing, and (2) the back end implementation shouldn't follow the awkward quirks of the front end. 

I think there's just enough surface area, now, to warrant a PatientPreferencesService, and for the front end to interact with these values by querying and mutating the PatientPreferences entity directly in the relevant areas.